test-data/unit/check-warnings.test

```diff
@@ -145,8 +145,103 @@ main:4: warning: Returning Any from function declared to return "int"
 
 [case testReturnAnyForNotImplementedInBinaryMagicMethods]
 # flags: --warn-return-any
+from typing import Any
+class A:
+    def __init__(self, x: Any) -> None:
+        self.x = x
+    def __eq__(self, other: Any) -> bool:
+        if self.__class__ is other.__class__:
+            return bool(self.x == other.x)
+        return NotImplemented
+    def __lt__(self, other: Any) -> bool: return NotImplemented
+    def __gt__(self, other: Any) -> bool: return NotImplemented
+    def __le__(self, other: Any) -> bool: return NotImplemented
+    def __ge__(self, other: Any) -> bool: return NotImplemented
+    def __ne__(self, other: Any) -> bool:
+        # The logic here is similar to __eq__, except the
+        # "return" is inside the if-then-else. The intention
+        # is to also verify that both ways of inferring
+        # Union[bool, type(NotImplemented)] are checked.
+        if self.__class__ is other.__class__:
+            return bool(self.x != other.x)
+        else:
+            return NotImplemented
+# TODO: If the programmer specifies "-> Union[bool, type(NotImplemented)]",
```

Comment:
The need for all these comments here still makes me unhappy, and nothing seems to answer the question why `__ne__` is special.